File: Test/baiGiang/array/sinh.cpp
#include <bits/stdc++.h>

using namespace std;

mt19937_64 rng(chrono::high_resolution_clock::now().time_since_epoch().count());
int randint(int l, int r) { return uniform_int_distribution<int>(l, r)(rng); }

signed main() {
  srand(time(0));
  system("g++ chuan.cpp -o chuan");
  system("g++ trau.cpp -o trau");

  int limTest = 1313;
  int lim = 1000;
  for (int test = 1; test <= limTest; test++) {
    // Generate test file
    ofstream inp("TEST.INP");
    int n = randint(1, lim);
    int k = randint(1, lim);
    inp << n << " " << k << "\n";
    for (int i = 0; i < n; i++) {
      if (randint(1, lim) % 5 == 0) {
        inp << k << " ";
        continue;
      }
      inp << randint(1, lim) << " ";
    }
    inp.close();

    // Checker
    cout << "--- RUNNING ON TEST " << test << " ---\n";
    cout << "RESULT: ";
    system("./chuan");
    system("./trau");

    int check = system("diff CHUAN.OUT TRAU.OUT");
    if (check) {
      cout << "WA!!!\n\n";
      return 0;
    }
    cout << "AC\n\n";
  }

  cout << "---CONGRATULATION!!!---\n";
  cout << "ALL TEST PASSED SUCCESSFULLY\n\n";
}

File: Test/baiGiang/array/sinh.py
from subprocess import run, TimeoutExpired, CalledProcessError
from random import randint

LIM = 100000
TEST = 100
CHUAN = "chuan"
TRAU = "trau"


def gen(lim):
    n = randint(1, lim)
    k = randint(1, lim)
    a = [str(randint(1, lim)) for _ in range(n)]
    with open("test.inp", "w") as f:
        f.write(f"{n} {k}\n")
        f.write(" ".join(a))


def exc(prog, time):
    try:
        run([f"./{prog}"], timeout=time, check=1)
    except TimeoutExpired:
        print(f"[TLE] - {prog}")
        exit()
    except CalledProcessError as e:
        print("f[RE] - {prog}. ERROR CODE:")
        print(e)
        exit()


run(["g++", f"{CHUAN}.cpp", "-o", f"{CHUAN}"])
run(["g++", f"{TRAU}.cpp", "-o", f"{TRAU}"])

for test in range(TEST):
    print(f"TEST {test + 1}: ", end="")
    gen(LIM)

    exc(CHUAN, 1)
    exc(TRAU, 1)

    with open(f"{CHUAN}.out", "r") as f:
        chuan = f.read()

    with open(f"{TRAU}.out", "r") as f:
        trau = f.read()

    if chuan != trau:
        print("[WA]")
        exit()
    else:
        print("[AC]")
